main/lsp/watchman/WatchmanProcess.cc

```diff
@@ -21,6 +21,17 @@ WatchmanProcess::~WatchmanProcess() {
     // Destructor of Joinable ensures Watchman thread exits before this destructor finishes.
 };
 
+namespace {
+template <typename F> void catchDeserializationError(spdlog::logger &logger, const string &line, F &&f) {
+    try {
+        f();
+    } catch (sorbet::realmain::lsp::DeserializationError e) {
+        // Gracefully handle deserialization errors, since they could be our fault.
+        logger.error("Unable to deserialize Watchman request: {}\nOriginal request:\n{}", e.what(), line);
+    }
+}
+} // namespace
```

Comment:
I think I might have just wrapped the loop body or the `if/else` statement inside the loop in a single `try/catch` but this is fine too.